assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -3215,38 +3252,173 @@ protected String build_instructor_grade_submission_context(VelocityPortlet portl
                     checkForUsersInMultipleGroups(a, s.getSubmitterIds(), state, rb.getString("group.user.multiple.warning"));
                 }
 		
-		// for the navigation purpose
-		List<SubmitterSubmission> userSubmissions = state.getAttribute(USER_SUBMISSIONS) != null ? (List<SubmitterSubmission>) state.getAttribute(USER_SUBMISSIONS):null;
+		// SAK-29314
+		// Since USER_SUBMISSIONS is restricted to the page size, it is not very useful here
+		// Therefore, we will prefer to use STATE_PAGEING_TOTAL_ITEMS. However, sometimes this contains
+		// Assignment objects instead of SubmitterSubmission objects, so we have to fall back to USER_SUBMISSIONS
+		// if this occurs, although in practise this seems to never happen
+		List<SubmitterSubmission> userSubmissions = Collections.EMPTY_LIST;
+		Object totalItems = state.getAttribute(STATE_PAGEING_TOTAL_ITEMS);
+		if (totalItems != null)
```

Comment:
Casting nulls doesn't throw an exception, so this could be reduced with: 
List totalItems = (List)state.getAttribute(STATE_PAGEING_TOTAL_ITEMS);
if (!CollectionUtils.isEmpty(totalItems)) // requires import of apache's CollectionUtils, or just do '!= null && !.isEmpty'
{
if (totalItems.get(0) instanceof SubmitterSubmission)
...